src/GraphQL.MicrosoftDI/GraphQLBuilderExtensions.cs

```diff
@@ -0,0 +1,72 @@
+    {
+        /// <summary>
+        /// Returns a new <see cref="IGraphQLBuilder"/> for the specified service collection and
+        /// registers a default set of services required by GraphQL if they have not already been registered.
+        /// <br/><br/>
+        /// Does not include <see cref="IDocumentWriter"/>, and the default <see cref="IDocumentExecuter"/>
+        /// implementation does not support subscriptions.
+        /// <br/><br/>
+        /// Also configures <see cref="ComplexityConfiguration"/> to be pulled from the dependency
+        /// injection provider, overwriting values within <see cref="ExecutionOptions.ComplexityConfiguration"/>
+        /// with values configured within the registered instance if set there.
+        /// </summary>
+        public static IGraphQLBuilder AddGraphQL(this IServiceCollection services)
+            => new GraphQLBuilder(services);
+
+        /// <summary>
+        /// Registers <typeparamref name="TSchema"/> within the dependency injection framework. <see cref="ISchema"/> is also
+        /// registered if it is not already registered within the dependency injection framework. Services required by
+        /// <typeparamref name="TSchema"/> are instianted directly if not registered within the dependency injection framework.
+        /// This can eliminate the need to register each of the graph types with the dependency injection framework, either
+        /// manually or via <see cref="GraphQL.GraphQLBuilderExtensions.AddGraphTypes(IGraphQLBuilder)"/>. Singleton and scoped
+        /// lifetimes are supported.
+        /// </summary>
+        /// <remarks>
+        /// Schemas that implement <see cref="IDisposable"/> of a transient lifetime are not supported, as this will cause a
+        /// memory leak if requested from the root service provider.
+        /// </remarks>
+        public static IGraphQLBuilder AddSelfActivatingSchema<TSchema>(this IGraphQLBuilder builder, ServiceLifetime serviceLifetime = ServiceLifetime.Singleton)
+            where TSchema : class, ISchema
+        {
+            if (serviceLifetime == ServiceLifetime.Transient && typeof(IDisposable).IsAssignableFrom(typeof(TSchema)))
+            {
+                // This scenario can cause a memory leak if the schema is requested from the root service provider.
+                // If it was requested from a scoped provider, then there is no reason to register it as transient.
+                // See following link:
+                // https://docs.microsoft.com/en-us/dotnet/core/extensions/dependency-injection-guidelines#disposable-transient-services-captured-by-container
+                throw new InvalidOperationException("A schema that implements IDisposable cannot be registered as a transient service.");
+            }
+
+            // Register the service with the DI provider as TSchema, overwriting any existing registration
+            builder.Register(services =>
```

Comment:
services  -> provider here and for line 62.

Let's use naming convention that _services_ is for `IServiceCollection` and _provider_ is for `IServiceProvider`.

Reply:
I do not mind in this file, as `IServiceCollection` and `IServiceProvider` are both used in this same file.  (I feel that typically `IServiceProvider` is named `services`.)

#2785